db/migrate/20200331192627_assign_names_to_users_that_bypassed_profile_creation.rb

```diff
@@ -0,0 +1,12 @@
+class AssignNamesToUsersThatBypassedProfileCreation < ActiveRecord::Migration[5.2]
+  def up
+    User.joins(:application_users).where(state: :needs_profile).distinct.find_each do |user|
+      user.first_name ||= 'Unknown'
+      user.last_name ||= user.email_addresses.verified.first&.value || 'Unknown'
```

Comment:
I don't think this is what Carly meant. I think users may have either 1) a username or 2) first and last name. So what Carly and I were suggesting is that you just generate a `username` for them based on their email address. So for me, for example, my email address is `bryaneli@rice.edu`—make my username `bryaneli`? Looks better than first name `Unknown`.

Reply:
I'll let @carlycurran decide what to do here.

Reply:
Yes I was thinking that we could leave first and last names blank and make the username filed the email address. Does that conflict with the new accounts flow, where we only have email address and no user name?

Reply:
Accounts and the accounts-rails gem only allow letters, numbers and _ in the username so to put the email addresses in there we'd have to either replace the @ and . with _ or relax these rules.

Reply:
ugh of course there's a catch. could we use what comes before the @ in the email address?

Reply:
replace all non_allowed w/ underscore?

Reply:
Done